wizard/vote_on_onchain_governance.py

```diff
@@ -0,0 +1,208 @@
+
+    close_p = sub.add_parser("close", help="close a proposal")
+    close_p.add_argument("--proposal", type=int, required=True, help="proposal nonce / id")
+
+    propose_p = sub.add_parser("propose", help="create a new proposal")
+    propose_p.add_argument("--commit-hash", required=True, help="commit hash for proposal content")
+    propose_p.add_argument("--start-epoch", type=int, required=True, help="start vote epoch")
+    propose_p.add_argument("--end-epoch", type=int, required=True, help="end vote epoch")
+    propose_p.add_argument("--fee-egld", type=float, default=0.0, help="native EGLD amount to attach")
+
+    clear_p = sub.add_parser("clear", help="clear ended proposals for given proposers")
+    clear_p.add_argument("--proposers", nargs="+", required=True, help="list of bech32 proposer addresses")
+
+    claim_p = sub.add_parser("claim-fees", help="claim accumulated governance fees")
+
+    chgcfg_p = sub.add_parser("change-config", help="change governance configuration")
+    chgcfg_p.add_argument("--proposal-fee", type=int, required=True, help="proposal fee (wei)")
+    chgcfg_p.add_argument("--lost-proposal-fee", type=int, required=True, help="lost proposal fee (wei)")
+    chgcfg_p.add_argument("--min-quorum", type=int, required=True, help="minimum quorum")
+    chgcfg_p.add_argument("--min-veto-threshold", type=int, required=True, help="minimum veto threshold")
+    chgcfg_p.add_argument("--min-pass-threshold", type=int, required=True, help="minimum pass threshold")
+
+    args = parser.parse_args(cli_args)
+
+    network = args.network
+    configuration = CONFIGURATIONS[network]
+    entrypoint = MyEntrypoint(configuration)
+
+    proxy_url = getattr(configuration, "proxy", None) or getattr(configuration, "proxy_url", None)
+    if not proxy_url:
+        raise errors.KnownError("CONFIGURATION is missing 'proxy' URL.")
+    provider = ProxyNetworkProvider(proxy_url)
+
+    chain_id = getattr(configuration, "chain_id", None)
+    if not chain_id:
+        netcfg = provider.get_network_config()
+        chain_id = getattr(netcfg, "chain_id", None)
+        if not chain_id:
+            raise errors.KnownError("Unable to resolve chain_id from provider network config.")
+
+    cfg = TransactionsFactoryConfig(chain_id=chain_id)
+
+    optional_limits = {
+        "gas_limit_for_vote": getattr(configuration, "gas_limit_for_vote", 6_000_000),
+        "gas_limit_for_proposal": getattr(configuration, "gas_limit_for_proposal", 12_000_000),
+        "gas_limit_for_closing_proposal": getattr(configuration, "gas_limit_for_closing_proposal", 6_000_000),
+        "gas_limit_for_clear_proposals": getattr(configuration, "gas_limit_for_clear_proposals", 6_000_000),
+        "gas_limit_for_claim_accumulated_fees": getattr(configuration, "gas_limit_for_claim_accumulated_fees", 6_000_000),
+        "gas_limit_for_change_config": getattr(configuration, "gas_limit_for_change_config", 12_000_000),
```

Comment:
this is not needed because the sdk takes care of setting the gas limit. we can simply initialize the `GasLimitEstimator` component and pass it to the factory.